src/OrchardCore/OrchardCore.Lucene.Core/QueryProviders/SimpleQueryStringQueryProvider.cs

```diff
@@ -17,7 +19,18 @@ public Query CreateQuery(ILuceneQueryService builder, LuceneQueryContext context
             var queryString = query["query"]?.Value<string>();
             var fields = query["fields"]?.Values<string>() ?? new string[0];
             var defaultOperator = query["default_operator"]?.Value<string>().ToLowerInvariant() ?? "or";
-            var weights = fields.ToDictionary(field => field, field => 1.0f);
+            float weight = 1.0f;
+            var weights = fields.ToDictionary(field => field, field => weight);
+
+            foreach(var field in fields)
+            {
+                if(field.Contains("^") && Single.TryParse(field.Split("^").Last(), out weight))
```

Comment:
Could probably use `StringComparison.Ordinal` in the `Contains`.

but looks good to me